logdetective/server/models.py

```diff
@@ -323,6 +331,63 @@ def __init__(self, data: Optional[dict] = None):
             self.instances[instance.url] = instance
 
 
+class KojiInstanceConfig(BaseModel):
+    """Model for Koji configuration of logdetective server."""
+
+    name: str = ""
+    xmlrpc_url: str = ""
+    max_artifact_size: int = 300 * 1024 * 1024
+    tokens: List[str] = []
+
+    _conn: Optional[koji.ClientSession] = None
+
+    def __init__(self, name: str, data: Optional[dict] = None):
+        super().__init__()
+
+        self.name = name
+        if data is None:
+            # Set some reasonable defaults
+            self.xmlrpc_url = "https://koji.fedoraproject.org/kojihub"
+            self.tokens = []
+            self.max_artifact_size = 1024 * 1024
+            return
+
+        self.xmlrpc_url = data.get(
+            "xmlrpc_url", "https://koji.fedoraproject.org/kojihub"
+        )
+        self.max_artifact_size = int(data.get("max_artifact_size")) * 1024 * 1024
+        self.tokens = data.get("tokens", [])
+
+    def get_connection(self):
+        """Get the Koji connection object"""
+        if not self._conn:
+            self._conn = koji.ClientSession(self.xmlrpc_url)
+        return self._conn
+
+
+class KojiConfig(BaseModel):
+    """Model for Koji configuration of logdetective server."""
+
+    instances: Dict[str, KojiInstanceConfig] = {}
+    analysis_timeout: int = 15
+
+    def __init__(self, data: Optional[dict] = None):
+        super().__init__()
+        if data is None:
+            return
+
+        if "analysis_timeout" in data:
+            self.analysis_timeout = data["analysis_timeout"]
+            del data["analysis_timeout"]
+        else:
+            self.analysis_timeout = 15
+
+        for instance_name, instance_data in data.items():
+            self.instances[instance_name] = KojiInstanceConfig(
```

Comment:
We could avoid this by putting configuration for individual instances into a list of its own, accessible by `koji_instances` key. That way the conditional and `del` on line 381 wouldn't be necessary.